docs/topics/data-analysis/data-analysis-work-with-data-sources.md

```diff
@@ -0,0 +1,187 @@
+[//]: # (title: Retrieve data from files)
+
+Kotlin Notebook, coupled with the [Kotlin DataFrame library](https://kotlin.github.io/dataframe/gettingstarted.html), enables 
+you to work with both non-structured and structured data. This combination offers the flexibility to transform non-structured data, 
+such as data found in TXT files, into structured datasets. 
+
+For data transformations, you can utilize methods such as [`add`](https://kotlin.github.io/dataframe/adddf.html), [`split`](https://kotlin.github.io/dataframe/split.html),
+[`convert`](https://kotlin.github.io/dataframe/convert.html), and [`parse`](https://kotlin.github.io/dataframe/parse.html). 
+Additionally, this toolset enables the retrieval and manipulation of data from various structured file formats, 
+including CSV, JSON, XLS, XLSX, and Apache Arrow.
+
+In this guide, you can learn how to retrieve, refine, and handle data through multiple examples.
+
+## Before you start
+
+1. Download and install the latest version of [IntelliJ IDEA Ultimate](https://www.jetbrains.com/idea/download/?section=mac).
+2. Install the [Kotlin Notebook plugin](https://plugins.jetbrains.com/plugin/16340-kotlin-notebook) in IntelliJ IDEA.
+
+   > Alternatively, access the Kotlin Notebook plugin from **Settings/Preferences** | **Plugins** | **Marketplace** within IntelliJ IDEA.
+   >
+   {type="tip"}
+
+3. Create a new Kotlin Notebook by selecting **File** | **New** | **Kotlin Notebook**.
+4. In the Kotlin Notebook, import the Kotlin DataFrame library by running the following command:
+
+```kotlin
+%use dataframe
+```
+
+## Retrieve data from a file
+
+To retrieve data from a file in Kotlin Notebook:
+
+1. Open your Kotlin Notebook file (`.ipynb`).
+2. Import the Kotlin DataFrame library by adding `%use dataframe` in a code cell at the start of your Notebook.
+> Make sure to run the code cell with the `%use dataframe` line before you run any other code cells that rely on the Kotlin DataFrame library.
+>
+{type="note"}
+
+3. Use the `.read()` function from the Kotlin DataFrame library to retrieve data. For example, to read a CSV file, 
```

Comment:
as far as I know, it is - I kept to this style because that's how it was written in the DataFrame library docs. "The .read() function automatically detects the input format based on a file extension and content:"